build.rs

```diff
@@ -168,13 +168,17 @@ fn main() {
         println!("cargo:rustc-cfg=portable_atomic_armv5te");
     }
 
-    // aarch64_target_feature stabilized in Rust 1.61.
-    // aarch64 macos always support lse: https://github.com/rust-lang/rust/blob/1.59.0/compiler/rustc_target/src/spec/aarch64_apple_darwin.rs#L5
-    if aarch64
-        && (version.minor >= 59 || version.nightly)
-        && has_target_feature("lse", target == "aarch64-apple-darwin", &version, Some(61))
-    {
-        println!("cargo:rustc-cfg=portable_atomic_target_feature_lse");
+    // aarch64 macos always support lse and lse2 because it is armv8.6: https://github.com/rust-lang/rust/blob/1.59.0/compiler/rustc_target/src/spec/aarch64_apple_darwin.rs#L5
+    if aarch64 && (version.minor >= 59 || version.nightly) {
+        // aarch64_target_feature stabilized in Rust 1.61.
+        if has_target_feature("lse", target == "aarch64-apple-darwin", &version, Some(61), true) {
+            println!("cargo:rustc-cfg=portable_atomic_target_feature_lse");
+        }
+        // As of rustc nightly-2022-04-30, target_feature "lse2" is not available on rustc side:
+        // https://github.com/rust-lang/rust/blob/d201c812d40932509b2b5307c0b20c1ce78d21da/compiler/rustc_codegen_ssa/src/target_features.rs#L45
```

Comment:
LLVM (and std_detect) already support it, so using it is not a problem, but we need to use [a hack in the build script](https://github.com/taiki-e/portable-atomic/blob/f5cc6fb235f2aa48aaf3a6563fd4de7171223148/build.rs#L340-L349).